web3/main.py

```diff
@@ -152,13 +153,53 @@ def currentProvider(self):
         ))
         return self.manager.providers[0]
 
-    @coerce_return_to_text
-    def sha3(self, value, encoding="hex"):
-        if encoding == 'hex':
-            hex_string = value
+    @staticmethod
+    @apply_to_return_value(encode_hex)
+    def sha3(primitive=None, text=None, hexstr=None, encoding=None):
+        if encoding is not None:
+            warnings.warn(DeprecationWarning(
+                "The encoding keyword has been deprecated.  Please update your "
+                "code to use sha3(text='txt'), sha3(hexstr='0x747874'), "
+                "sha3(b'\\x74\\x78\\x74'), or sha3(0x747874)."
+            ))
         else:
-            hex_string = encode_hex(value)
-        return self.manager.request_blocking('web3_sha3', [hex_string])
+            if not isinstance(primitive, (bytes, int, type(None))):
```

Comment:
`elif not ....`

Looks like you have one level of unnecessary indentation here.